dir.props

```diff
@@ -71,6 +72,10 @@
   </PropertyGroup>
   <Import Project="$(ToolRuntimePath)/net45/roslyn/build/Microsoft.Net.Compilers.props" Condition="'$(UseRoslynCompilers)'!='false'" />
 
+  <PropertyGroup>
+    <_BuildAgainstPackages Condition="'$(KeepAllProjectReferences)' != 'true' and $([System.Text.RegularExpressions.Regex]::IsMatch('$(MSBuildProjectFullPath)', '\\tests\\', System.Text.RegularExpressions.RegexOptions.IgnoreCase))">$(BuildTestsAgainstPackages)</_BuildAgainstPackages>
```

Comment:
Ah, I see from your comment below that build.common.targets is likely a better location so that I can move properties like "KeepAllProjectReferences" after the dir.props import.